hbase-server/src/test/java/org/apache/hadoop/hbase/master/balancer/BalancerTestBase.java

```diff
@@ -174,6 +174,46 @@ public void reloadCachedMappings(List<String> arg0) {
     }
   }
 
+  // This mock allows us to test the LocalityCostFunction
+  public class MockCluster extends BaseLoadBalancer.Cluster {
+
+    private int[][] localities = null;   // [region][server] = percent of blocks
+    private boolean firstRegionAsReplica;
+
+    public MockCluster(int[][] regions) {
+      this(regions, false);
+    }
+
+    public MockCluster(int[][] regions, final boolean firstRegionAsReplica) {
```

Comment:
This mocking is completely inscrutable to me, but I guess you simply moved it from `TestStochasticLoadBalancer`.

Reply:
Yeah, this is just simply moved from the TestStochasticLoadBalancer.java (if I am not wrong), cause it is needed for the new test case TestStochasticLoadBalancerRegionReplica.